### ISSUE TITLE
Strange output when training Java model

### ISSUE THREAD

**Issue by tkswanson** [conservationtechlab/animl-py#93]
Poor performance, slow epochs, no caching, no cropping, etc

**Comment by matobler** [comment]
Issues were found both with the training data preparation and training pipeline. This lead to the following changes and improvements:

Date preparation changes:

- Only use MD class 1 (animals) for training. Removed human and vehicle from model.
- Only use crops with a MD conf >0.5
- Reduced validation set to 200 images/species or 20% of images, whichever is smaller (needs further evaluation).
- Ensured that no images are in the training and validation set through an explicit check.

Training pipeline changes:

- Added option for AdamW optimizer
- Switched rate scheduler to CosineAnnealingLR
- Added local caching of cropped images (significant increase in training speed)
- Added option for mix-precision training (increase in training speed, reduced usage of VRAM)
- Saving and loading of checkpoints now includes optimizer and scheduler states
- Added additional parameters in config file (momentum, t_max, etc.)
- Added a log file to track loss, accuracy and learning rate
- Added a separate batch size for validation
- Set Crop to default to True
